### DISCUSSION TITLE
Renaming placeholder labels

### DISCUSSION THREAD

**Discussion by kathrine-mcaulay** [davidgohel/officer#587] [Q&A] [answered]
My institutional powerpoint template has many duplicate placeholder labels. It looks like my two options are: 1) Use the location IDs instead or 2) Manually relabel these in the template before importing. I'd love to avoid either of these as there are 87 placeholders throughout the template, 20 of which are not unique at the layout level.

I can modify the `ph_label` variable in the dataframe generated by `layout_properties()`; however, looking at the way the template is stored when read in via `read_pptx()`, I think it's doubtful that the placeholder labels themselves can be modified from r, but I'm asking anyway, in case someone has found a convenient workaround for a similar situation.

**Comment by markheckmann** [comment, marked as answer]
Hi @kathrine-mcaulay,

the development version now has a new function `layout_dedupe_ph_labels()` which partially does what you want. While it does not allow custom renaming of duplicate placeholder labels, it will automatically create unique ph labels. See  #589.

Note that this version has not yet been released to CRAN. You can install the development version to see if it works for you.

```r
devtools::install_github("davidgohel/officer")
```

**Reply by kathrine-mcaulay** [reply]
This is perfect! At least in my case, it's not essential to specify the names, just to distinguish them, which is exactly what this does. This worked well in a quick test. Thank you!
